features/steps/steps.py

```diff
@@ -201,6 +253,54 @@ def step_impl(context, constraint, num, entity):
 
     handle_errors(context, errors)
 
+@then('The element must have unique values for the attribute {}')
+def step_impl(context, attribute):
+    errors = []
+    unwrap = list(filter(None, list(map(lambda layer: layer.get('entities'), context._stack))))
+    if len(unwrap):
+        last = unwrap[-1].get('var')
+
+        for instances in context.instances:
+            attribute_instances = instances
+            instance_value_pair = list(map(lambda i: (i, getattr(i, attribute, random_string())), attribute_instances))
+            duplicate_values = [value for (value,count) in Counter([v[1] for v in instance_value_pair]).items() if count > 1]
+            if len(duplicate_values):
+                false_instances = [i[0] for i in instance_value_pair if i[1] in duplicate_values]
+                errors.append(representation_value_error(last, duplicate_values, false_instances))
+    handle_errors(context, errors)
+
+
+@then('The values for attribute {attribute} shall be unique')
+def step_impl(context, attribute):
+    errors = []
+    for inst in context.instances:
+        attribute_instances = inst[1]
+        instance_value_pair = list(map(lambda i: (i, getattr(i, attribute, random_string())), attribute_instances))
+        duplicate_values = [value for (value,count) in Counter([v[1] for v in instance_value_pair]).items() if count > 1]
+        if len(duplicate_values):
+            false_instances = [i[0] for i in instance_value_pair if i[1] in duplicate_values]
+            errors.append(representation_value_error(inst[0], duplicate_values, false_instances))
+
+    handle_errors(context, errors)
+
+
+@then('Each instance of {entity} has a unique value for the attribute {attribute}')
+def step_impl(context, entity, attribute):
+    ent_attr = {'IfcShapeRepresentation':'Representations'}
+    assert entity in ent_attr
+    attr_get = ent_attr[entity]
+
+    errors = []
+    for inst in context.instances:
+        attribute_intances = getattr(inst, attr_get, [])
+        instance_value_pair = list(map(lambda i: (i, getattr(i, attribute, random_string())), attribute_intances))  #[instance, value),(i,v),..]
+        duplicate_values = [value for (value,count) in Counter([v[1] for v in instance_value_pair]).items() if count > 1]
+        if len(duplicate_values):
+            false_instances = [i[0] for i in instance_value_pair if i[1] in duplicate_values]
+            errors.append(representation_value_error(inst, duplicate_values, false_instances))
+
+    handle_errors(context, errors)
```

Comment:
This _then_ is commented out - should the function remain in code? Is it called anywhere?